### ISSUE TITLE
a needless slash returned by show-option cmd

### ISSUE THREAD

**Issue by zhangjiakouzf** [tmux/tmux#3946]
### Issue description
In ~/.tmux.conf, I define a global variable @resurrect-dir  with the following statement:
        set -g @resurrect-dir '$HOME/.tmux/resurrect' 
but when I read it with "show-option" cmd, the return string begins with a needless slash
$./tmux show-option -gqv "@resurrect-dir"
\\$HOME/.tmux/resurrect

this issue occurs in current version 
$./tmux -V
tmux next-3.5

but that's fine in 3.3a version which is installed by yum.

### Required information

Please provide the following information:

* tmux version (`tmux -V`).
   tmux next-3.5 
* Platform (`uname -sp`).
 Linux x86_64
* $TERM inside and outside of tmux (`echo $TERM`).
xterm-256color
* Logs from tmux (`tmux kill-server; tmux -vv new`).
1713381862.684818 /dev/pts/16: resurrect file:\\/home/admin/.tmux/resurrect/last
...
1713381862.687647 format_expand1: expanding format: Tmux resurrect file not found!
1713381862.687649 format_expand1: result is: Tmux resurrect file not found!
1713381862.687652 status_message_set: Tmux resurrect file not found!                                                                       1713381862.687656 message: /dev/pts/16 message: Tmux resurrect file not found!

![image](https://github.com/tmux/tmux/assets/1612026/e8c54de8-1cf4-4c35-a5df-d1697c3c5065)

git diff 0b355ae8114511e1ff6359272b164f1cdf718e80 server-client.c
![image](https://github.com/tmux/tmux/assets/1612026/0c1317f9-acdf-4320-bb56-55c35e48815d)

**Comment by nicm** [comment]
This is probably a simple fix, since whether environment variables are escaped only matters in quotes:
~~~~~
--- a/utf8.c
+++ b/utf8.c
@@ -525,7 +525,7 @@ utf8_strvis(char *dst, const char *src, size_t len, int flag)
                        /* Not a complete, valid UTF-8 character. */
                        src -= ud.have;
                }
-               if (src[0] == '$' && src < end - 1) {
+               if ((flag & VIS_DQ) && src[0] == '$' && src < end - 1) {
                        if (isalpha((u_char)src[1]) ||
                            src[1] == '_' ||
                            src[1] == '{')

**Comment by rhelmot** [comment]
I can confirm that this patch fixes a similar issue I was having with tmux 3.4 on ubuntu 24.04.

reproducer:

```python
#!/usr/bin/env python3

# python script to make triple sure that there's no escaping being done by my shell
import subprocess
subprocess.run(['tmux', '-V'])
subprocess.run(['tmux', 'set-environment', 'ASDF', '"${WEH}"'])
subprocess.run(['tmux', 'show-environment', 'ASDF'])
```

Before:

```
tmux 3.4
ASDF="\${WEH}"
```

After:

```
tmux next-3.5
ASDF="${WEH}"
```

**Comment by nicm** [comment]
Applied to OpenBSD now, will be in GitHub later. Thanks!
